File: Fibonacci/01-fibonacci.py
count = 2


def fibonacci(num1, num2, termo):
    global count
    if count <= termo:
        num3 = num1 + num2
        print(num3)

        num1, num2 = num2, num3

        count += 1

        return fibonacci(num1, num2, termo)
    else:
        return


termo = int(input("> Digite quantos termos da sequência de fibonacci: "))

print(0)
print(1)

fibonacci(0, 1, termo)

File: Fibonacci/01-fibonacci.c
#include <stdio.h>

void fibonacci(int prev1, int prev2, int termo, int* count);

int main()
{

    int prev1 = 1, prev2 = 0;
    int count = 2;
    int termo;

    printf("Defina quantos termos a sequência de fibonacci terá: ");
    scanf("%d", &termo);

    printf("%d\n%d\n", prev2, prev1);

    fibonacci(prev1, prev2, termo, &count);

    return 0;
}

void fibonacci(int prev1, int prev2, int termo, int* count)
{
    if (*count <= termo)
    {
        int num = prev1 + prev2;

        printf("%d\n", num);

        prev2 = prev1;
        prev1 = num;

        (*count) += 1;

        fibonacci(prev1, prev2, termo, count);
    }
    else
    {
        return;
    }
}